src/lib/sources.js

```diff
@@ -153,53 +154,63 @@ class Sources extends EventEmitter {
    * @return {Object}
    */
   health() {
-    const object = {
+    const obj = {
       code: STATUS_CODES.OK,
-      status: 'OK'
+      status: 'OK',
+      indices: this.indices.map((i) => i.status()),
+      sources: this.properties.sources.map((s) => s.status())
     };
 
-    const source_state = {
-      source: [],
-      count: 0
-    };
+    // Health logic:
+    // Indices:
+    // 500 - any index is !ok
+    // 503 - any index is !ready
+    // 200 - all indices are both ok and ready
+    // Sources:
+    // 500 - all sources are !ok
+    // 503 - any source is !ready
+    // 200 - all sources both ok and ready
+    //
+    const notReady = (s) => !s.ready;
+    const unhealthy = (s) => !s.ok;
+
+    if (this.indices.some(unhealthy)) {
+      const u = this.indices.find((i) => !i.ok);
+
+      obj.code = STATUS_CODES.INTERNAL_SERVER_ERROR;
+      obj.status = u.status().state;
+
+      return obj;
+    }
 
-    object.indices = this.indices.map((source) => {
-      // TODO This logic is fairly ham-fisted right now. It'll work because nothing
-      // is setting `state` to WARNING at the moment. When we do start supporting a
-      // WARNING state, this will have to become aware that OK < WARING < ERROR.
-      if (!source.ready) {
-        object.code = STATUS_CODES.SERVICE_UNAVAILABLE;
-        object.status = source.state;
-      }
+    if (this.indices.some(notReady)) {
+      const u = this.indices.find((i) => !i.ready);
 
-      if (!source.ok) {
-        object.code = STATUS_CODES.INTERNAL_SERVER_ERROR;
-        object.status = source.state;
-      }
+      obj.code = STATUS_CODES.SERVICE_UNAVAILABLE;
+      obj.status = u.status().state;
 
-      return source.status();
-    });
+      return obj;
+    }
 
-    object.sources = this.properties.sources.map((source) => {
-      if (!source.ready) {
-        object.code = STATUS_CODES.SERVICE_UNAVAILABLE;
-        object.status = source.state;
-      }
+    if (this.properties.sources.every(unhealthy)) {
```

Comment:
`every` does not work as intended here - probably safer with the current function
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every

Reply:
possibly need to add a `!`

Reply:
Ignore this - there was an issue with my local git branch for this PR after the force-push fix that was done and it does work

Reply:
You are right that there's an issue here. I need to do a sources length check because `every` returns true for empty arrays but no props because of no sources is still a "healthy" state even if it's not an expected use.